src/app/api/volunteer/auth/login/route.ts

```diff
@@ -0,0 +1,36 @@
+import { getUserByEmail } from "@server/mongodb/actions/User";
+import APIWrapper from "@server/utils/APIWrapper";
+import User from "@server/mongodb/models/User";
+import { IUser, Role } from "@/common_utils/types";
+
+export const dynamic = "force-dynamic";
+
+export const GET = APIWrapper({
+  config: {
+    requireToken: true,
+  },
+
+  handler: async (req) => {
+    const { searchParams } = new URL(req.url);
+    const email = searchParams.get("email");
+    if (!email) {
+      throw new Error("Email parameter is missing in the request.");
+    }
+
+    let newUser = await getUserByEmail(email);
+    if (newUser === null) {
+      try {
```

Comment:
Since the APIWrapper catches MongoDB Errors, we do not need a custom try-catch block for user creation query